pkg/virt-controller/watch/descheduler/descheduler.go

```diff
@@ -19,7 +19,85 @@
 
 package descheduler
 
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/client-go/tools/cache"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/controller"
+
+	k8sv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+)
+
 // EvictOnlyAnnotation indicates pods whose eviction is not expected to be completed right away.
 // Instead, an eviction request is expected to be intercepted by an external component which will initiate the
 // eviction process for the pod.
 const EvictOnlyAnnotation = "descheduler.alpha.kubernetes.io/request-evict-only"
+
+// EvictionInProgressAnnotation indicates pods whose eviction was initiated by an external component.
+const EvictionInProgressAnnotation = "descheduler.alpha.kubernetes.io/eviction-in-progress"
+
+func AddDeschedulerAnnotation(virtClient kubecli.KubevirtClient, sourcePod *k8sv1.Pod) error {
+	if _, exists := sourcePod.GetAnnotations()[EvictionInProgressAnnotation]; exists {
+		return nil
+	}
+
+	patchSet := patch.New(
+		patch.WithAdd(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(EvictionInProgressAnnotation)), ""),
+	)
+	patchBytes, err := patchSet.GeneratePayload()
+	if err != nil {
+		return err
+	}
+
+	_, err = virtClient.CoreV1().Pods(sourcePod.Namespace).Patch(context.Background(), sourcePod.Name, types.JSONPatchType, patchBytes, v1.PatchOptions{})
+	if err != nil {
+		log.Log.Object(sourcePod).Errorf("failed to add %s pod annotation: %v", EvictionInProgressAnnotation, err)
+		return err
+	}
+
+	return nil
+}
+
+func RemoveDeschedulerAnnotationFromSourcePod(virtClient kubecli.KubevirtClient, migration *virtv1.VirtualMachineInstanceMigration, podIndexer cache.Indexer) error {
+	if migration.Status.MigrationState == nil || migration.Status.MigrationState.SourcePod == "" {
+		return nil
+	}
+
+	podKey := controller.NamespacedKey(migration.Namespace, migration.Status.MigrationState.SourcePod)
+	obj, exists, err := podIndexer.GetByKey(podKey)
+	if !exists {
+		log.Log.Warningf("source pod %s does not exist", migration.Status.MigrationState.SourcePod)
+		return nil
+	}
+	if err != nil {
+		log.Log.Reason(err).Errorf("Failed to fetch source pod %s for namespace from cache.", migration.Status.MigrationState.SourcePod)
+		return err
+	}
+
+	sourcePod := obj.(*k8sv1.Pod)
+	if _, exists := sourcePod.GetAnnotations()[EvictionInProgressAnnotation]; exists {
+		patchSet := patch.New(patch.WithRemove(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(EvictionInProgressAnnotation))))
```

Comment:
I'd do a test + remove for the same reason as above.

Reply:
And the discussion continues :) https://github.com/kubevirt/kubevirt/pull/12025#discussion_r1629204451